soc/arm/microchip_mec/mec172x/reg/mec172x_regs.h

```diff
@@ -0,0 +1,1135 @@
+	struct espi_smvw_reg SMVW[11];
+};
+
+/** @brief HW implements 11 Target-to-Host VW registers as named registers */
+struct espi_smvw_named_regs {
+	struct espi_smvw_reg SMVW00;
+	struct espi_smvw_reg SMVW01;
+	struct espi_smvw_reg SMVW02;
+	struct espi_smvw_reg SMVW03;
+	struct espi_smvw_reg SMVW04;
+	struct espi_smvw_reg SMVW05;
+	struct espi_smvw_reg SMVW06;
+	struct espi_smvw_reg SMVW07;
+	struct espi_smvw_reg SMVW08;
+	struct espi_smvw_reg SMVW09;
+	struct espi_smvw_reg SMVW10;
+};
+
+/** @brief eSPI S2M VW registers as an array of words at 0x400F9E00 */
+struct espi_smvw32_regs {
+	volatile uint32_t SMVW[11 * 2];
+};
+
+/* eSPI SAF */
+/** @brief SAF SPI Opcodes and descriptor indices */
+struct mchp_espi_saf_op {
+	volatile uint32_t OPA;
+	volatile uint32_t OPB;
+	volatile uint32_t OPC;
+	volatile uint32_t OP_DESCR;
+};
+
+/** @brief SAF protection regions contain 4 32-bit registers. */
+struct mchp_espi_saf_pr {
+	volatile uint32_t START;
+	volatile uint32_t LIMIT;
+	volatile uint32_t WEBM;
+	volatile uint32_t RDBM;
+};
+
+/** @brief eSPI SAF configuration and control registers at 0x40008000 */
+struct mchp_espi_saf {
+	uint32_t RSVD1[6];
+	volatile uint32_t SAF_ECP_CMD;
+	volatile uint32_t SAF_ECP_FLAR;
+	volatile uint32_t SAF_ECP_START;
+	volatile uint32_t SAF_ECP_BFAR;
+	volatile uint32_t SAF_ECP_STATUS;
+	volatile uint32_t SAF_ECP_INTEN;
+	volatile uint32_t SAF_FL_CFG_SIZE_LIM;
+	volatile uint32_t SAF_FL_CFG_THRH;
+	volatile uint32_t SAF_FL_CFG_MISC;
+	volatile uint32_t SAF_ESPI_MON_STATUS;
+	volatile uint32_t SAF_ESPI_MON_INTEN;
+	volatile uint32_t SAF_ECP_BUSY;
+	uint32_t RSVD2[1];
+	struct mchp_espi_saf_op SAF_CS_OP[2];
+	volatile uint32_t SAF_FL_CFG_GEN_DESCR;
+	volatile uint32_t SAF_PROT_LOCK;
+	volatile uint32_t SAF_PROT_DIRTY;
+	volatile uint32_t SAF_TAG_MAP[3];
+	struct mchp_espi_saf_pr SAF_PROT_RG[17];
+	volatile uint32_t SAF_POLL_TMOUT;
+	volatile uint32_t SAF_POLL_INTRVL;
+	volatile uint32_t SAF_SUS_RSM_INTRVL;
+	volatile uint32_t SAF_CONSEC_RD_TMOUT;
+	volatile uint16_t SAF_CS0_CFG_P2M;
+	volatile uint16_t SAF_CS1_CFG_P2M;
+	volatile uint32_t SAF_FL_CFG_SPM;
+	volatile uint32_t SAF_SUS_CHK_DLY;
+	volatile uint16_t SAF_CS0_CM_PRF;
+	volatile uint16_t SAF_CS1_CM_PRF;
+	volatile uint32_t SAF_DNX_PROT_BYP;
+};
+
+struct mchp_espi_saf_comm { /* @ 0x40071000 */
+	uint32_t TEST0;
+	uint32_t TEST1;
+	uint32_t TEST2;
+	uint32_t TEST3;
+	uint32_t TEST4;
+	uint32_t TEST5;
+	uint32_t TEST6;
+	uint32_t RSVD1[(0x2b8 - 0x01c) / 4];
+	uint32_t SAF_COMM_MODE; /* @ 0x400712b8 */
+	uint32_t TEST7;
+};
+
+/* Peripheral and SRAM base address */
+#define CODE_SRAM_BASE		0x000c0000u
+#define DATA_SRAM_BASE		0x00118000u
+#define DATA_SRAM_SIZE		0x10000u
+#define PERIPH_BASE		0x40000000u
+
+#define PERIPH_ADDR(ofs)	((uint32_t)PERIPH_BASE + (uint32_t)(ofs))
+
+#define PERIPH_INST_ADDR(ofs, inst, size) \
+	(PERIPH_ADDR(ofs) + (uint32_t)(inst) * (uint32_t)(size))
+
+/* Peripheral memory map */
+#define WDT_BASE		PERIPH_ADDR(0x400)
```

Comment:
@scottwcpg would these be eventually removed when full migration to device tree is in place?

Reply:
@albertofloyd Yes. I did an experiment last night adding most peripherals to device tree.  I was able to remove all the peripheral address defines from mec172x_regs.h  The mec172x DTSI now most devices. I am able to get addresses from DT in all the drivers, soc, and board level code.